File: min-platform-ui/src/utils/enums/oldRoleEnums.js
import { byName, byValue } from './enums'


export const dataScopeEnums = {
  ALL: {
    desc: "全部数据权限",
    name: "ALL"
  },
  CUSTOMIZE: {
    desc: "自定数据权限",
    name: "CUSTOMIZE"
  },
  CURRENT_DEPT: {
    desc: "本部门数据权限",
    name: "CURRENT_DEPT"
  },
  CURRENT_DEPT_AND_BELOW: {
    desc: "本部门及以下数据权限",
    name: "CURRENT_DEPT_AND_BELOW"
  },
  INDIVIDUAL: {
    desc: "个人数据",
    name: "INDIVIDUAL"
  },
  values() {
    return [
      this.ALL,
      this.CUSTOMIZE,
      this.CURRENT_DEPT,
      this.CURRENT_DEPT_AND_BELOW,
      this.INDIVIDUAL,
    ]
  },
  byName(name) {
    return byName(this.values(), name);
  },
};

export const roleStatusEnums = {
  ENABLE: {
    desc: '启用',
    value: true,
    name: 'ENABLE'
  },
  DISABLE: {
    desc: '禁用',
    value: false,
    name: 'DISABLE'
  },
  values() {
    return [
      this.ENABLE,
      this.DISABLE
    ]
  },
  byName(name) {
    return byName(this.values(), name)
  },
  byValue(value) {
    return byValue(this.values(), value)
  }
}

File: old-modules/old-module-business/src/main/java/com/old/business/user/enums/user/OldRoleEnums.java
package com.old.business.user.enums.user;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

/**
 * 角色表 实体的枚举类 old_role
 *
 * @author old
 * @date 2023-05-08
 */
public interface OldRoleEnums {


    @Getter
    @ToString
    @AllArgsConstructor
    enum DataScope {
        // 数据范围（1：全部数据权限 2：自定数据权限 3：本部门数据权限 4：本部门及以下数据权限）
        ALL("全部数据权限"),
        CUSTOMIZE("自定数据权限"),
        CURRENT_DEPT("本部门数据权限"),
        CURRENT_DEPT_AND_BELOW("本部门及以下数据权限"),
        INDIVIDUAL("个人数据"),
        ;

        private final String desc;

    }

}
